src/renderer/components/+helm-releases/release-details/release-details-model/release-details-model.injectable.tsx

```diff
@@ -0,0 +1,312 @@
+import navigateToHelmReleasesInjectable from "../../../../../common/front-end-routing/routes/cluster/helm/releases/navigate-to-helm-releases.injectable";
+import assert from "assert";
+import withOrphanPromiseInjectable from "../../../../../common/utils/with-orphan-promise/with-orphan-promise.injectable";
+
+const releaseDetailsModelInjectable = getInjectable({
+  id: "release-details-model",
+
+  instantiate: (di, targetRelease: TargetHelmRelease) => {
+    const callForHelmRelease = di.inject(callForHelmReleaseInjectable);
+    const callForHelmReleaseConfiguration = di.inject(callForHelmReleaseConfigurationInjectable);
+    const themeStore = di.inject(themeStoreInjectable);
+    const getResourceDetailsUrl = di.inject(getResourceDetailsUrlInjectable);
+    const updateRelease = di.inject(updateReleaseInjectable);
+    const showCheckedErrorNotification = di.inject(showCheckedErrorNotificationInjectable);
+    const showSuccessNotification = di.inject(showSuccessNotificationInjectable);
+    const createUpgradeChartTab = di.inject(createUpgradeChartTabInjectable);
+    const navigateToHelmReleases = di.inject(navigateToHelmReleasesInjectable);
+    const withOrphanPromise = di.inject(withOrphanPromiseInjectable);
+
+    const model = new ReleaseDetailsModel({
+      callForHelmRelease,
+      targetRelease,
+      themeStore,
+      callForHelmReleaseConfiguration,
+      getResourceDetailsUrl,
+      updateRelease,
+      showCheckedErrorNotification,
+      showSuccessNotification,
+      createUpgradeChartTab,
+      navigateToHelmReleases,
+    });
+
+    const load = withOrphanPromise(model.load);
+
+    // TODO: Reorganize Drawer to allow setting of header-bar in children to make "getPlaceholder" from injectable usable.
+    load();
+
+    return model;
+  },
+
+  lifecycle: lifecycleEnum.keyedSingleton({
+    getInstanceKey: (di, release: TargetHelmRelease) =>
+      `${release?.namespace}/${release?.name}`,
+  }),
+});
+
+export default releaseDetailsModelInjectable;
+
+export interface OnlyUserSuppliedValuesAreShownToggle {
+  value: IObservableValue<boolean>;
+  toggle: () => Promise<void>;
+}
+
+export interface ConfigurationInput {
+  nonSavedValue: IObservableValue<string>;
+  isLoading: IObservableValue<boolean>;
+  isSaving: IObservableValue<boolean>;
+  onChange: (value: string) => void;
+  save: () => Promise<void>;
+}
+
+interface Dependencies {
+  callForHelmRelease: CallForHelmRelease;
+  targetRelease: TargetHelmRelease;
+  themeStore: ThemeStore;
+  callForHelmReleaseConfiguration: CallForHelmReleaseConfiguration;
+  getResourceDetailsUrl: GetResourceDetailsUrl;
+  updateRelease: CallForHelmReleaseUpdate;
+  showCheckedErrorNotification: ShowCheckedErrorNotification;
+  showSuccessNotification: ShowNotification;
+  createUpgradeChartTab: (release: HelmRelease) => string;
+  navigateToHelmReleases: NavigateToHelmReleases;
+}
+
+export class ReleaseDetailsModel {
```

Comment:
Note: I'd call this view model since it is the only dependency that `ReleaseDetails` has and it combines all of the functionality and business logic details needs. It doesn't expose anything else than is required and it's allowed to be injected only by `ReleaseDetails`. Therefore it serves only one purpose.